salt/config/__init__.py

```diff
@@ -2848,14 +2848,14 @@ def is_profile_configured(opts, provider, profile_name, vm_=None):
     # If cloning on Linode, size and image are not necessary.
     # They are obtained from the to-be-cloned VM.
     if driver == 'linode' and profile_key.get('clonefrom', False):
```

Comment:
The lines below will execute only if driver is ``linode`` and ``clonefrom`` is specified in the profile which is when you would treat ``linode`` driver as a non image and non size driver